infer/src/integration/JsonReports.ml

```diff
@@ -140,6 +178,80 @@ end) : Printer with type elt = P.elt = struct
         ()
 end
 
+module type PrinterSarif = sig
+  type elt
+
+  val pp_open : F.formatter -> unit -> unit
+
+  val pp_results : F.formatter -> unit -> unit
+
+  val pp_close : F.formatter -> unit -> unit
+
+  val pp_rules : F.formatter -> elt -> unit
+
+  val pp : F.formatter -> elt -> unit
+end
+
+module MakeSarifListPrinter (P : sig
```

Comment:
This functor has only one instantiation, could it be simply a module instead? I.e. inline the functions in `PrinterSarif`.